utils/digital_interrupts.go

```diff
@@ -11,15 +12,46 @@ import (
 	"go.viam.com/rdk/resource"
 )
 
-// DigitalInterruptConfig describes the configuration of digital interrupt for the board.
-type DigitalInterruptConfig struct {
-	Name string `json:"name"`
-	Pin  string `json:"pin"`
-	Type string `json:"type,omitempty"` // e.g. basic, servo
+// PinConfig describes the configuration of a pin for the board
+type PinConfig struct {
```

Comment:
Didn't want to make another file for this, but I can if people would prefer this code to only have digital interrupt specific fields